apps/system/lockscreen/js/lockscreen_inputpad.js

```diff
@@ -144,6 +148,26 @@
     window.removeEventListener(name, cb);
   };
 
+
+  LockScreenInputpad.prototype.setPinLength = function(n) {
+    this.pinLength = n;
+
+    // If we've already called renderUI(), then update the UI.
+    if (this.passcodeCode) {
+      // Remove any existing content and add n <span> children
+      this.passcodeCode.textContent = '';
+      for(var i = 0; i < n; i++) {
+        this.passcodeCode.appendChild(document.createElement('span'));
+      }
+
+      // Allow CSS to display the n digits at the right size.
+      this.passcodeCode.dataset.pinLength = n;
+
+      // Set the right states on all of the spans
+      this.updatePassCodeUI();
+    }
```

Comment:
And I think it's part of the issue from we combined setter and rendering function into one.